eth/filters/dropped_tx_subscription.go

```diff
@@ -2,23 +2,74 @@ package filters
 
 import (
 	"context"
+	"time"
 
-	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/core"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/internal/ethapi"
 	"github.com/ethereum/go-ethereum/rpc"
 )
 
 type dropNotification struct {
-	TxHash common.Hash `json:"txhash"`
-	Reason string      `json:"reason"`
+	// TxHash common.Hash `json:"txhash"`
+	Tx          *ethapi.RPCTransaction `json:"tx"`
+	Reason      string                 `json:"reason"`
+	Replacement *ethapi.RPCTransaction `json:"replacedby,omitempty"`
+	Peer        interface{}            `json:"peer,omitempty"`
+	Time        int64                  `json:"ts"`
 }
 
 type rejectNotification struct {
 	Tx     *types.Transaction
 	Reason string `json:"reason"`
 }
 
+// newRPCTransaction returns a transaction that will serialize to the RPC
+// representation, with the given location metadata set (if available).
+func newRPCPendingTransaction(tx *types.Transaction) *ethapi.RPCTransaction {
+	if tx == nil {
+		return nil
+	}
+	var signer types.Signer
+	if tx.Protected() {
+		signer = types.LatestSignerForChainID(tx.ChainId())
+	} else {
+		signer = types.HomesteadSigner{}
+	}
+	from, _ := types.Sender(signer, tx)
```

Comment:
We need to handle this error.  LMK if it feels clunky -- there are patterns that make it easier.

Reply:
https://github.com/maticnetwork/bor/blob/87f9b91b254fd44264d1db353208a722fb818083/internal/ethapi/api.go#L1376 This function is essentially identical to how they're currently doing it in Bor and Geth, therefore I am hesitant to change it. It is also how it's being done in our go-ethereum mod.